packages/boot/test/integration/boot.integration.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: @loopback/boot
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect, Client, createClientForHandler} from '@loopback/testlab';
+import {NormalApplication} from '../fakeApp';
+import {Constructor, Binding} from '@loopback/context';
+import {BootMixin} from '../..';
+import {RestServer} from '@loopback/rest';
+import {
+  HelloController,
+  WorldController,
+} from '../fakeApp/controllers/hello.controller';
+import {AdminController} from '../fakeApp/controllers/admin/admin.controller';
+import {TestRepository} from '../fakeApp/repositories/test.repository';
+
+describe('@loopback/boot integration', () => {
+  class BootWithTestMixin extends TestMixin(BootMixin(NormalApplication)) {}
+  class BootWithTestMixinReverse extends BootMixin(
+    TestMixin(NormalApplication),
+  ) {}
+
+  let rootDir: string;
+
+  // Using any type for app so we can use the same stopApp function!
+  // tslint:disable-next-line:no-any
+  let app: any;
+  let client: Client;
+
+  before(getRootDir);
+  afterEach(stopApp);
+
+  it('booted controllers / app works as expected with TestMixin(BootMixin())', async () => {
```

Comment:
I think what the test is for should be more descriptive (testing for custom boot feat in this one, testing reversed order of mixins in the other, etc.), but maybe not since writing all of them would be too much; maybe just mention the important ones?